controllers/productController.ts

```diff
@@ -357,12 +359,19 @@ export const searchProductController: RequestHandler = async (req, res) => {
       error,
     });
   }
-};
+}) satisfies RequestHandler;
 
 // similar products
-export const realtedProductController: RequestHandler = async (req, res) => {
+export const realtedProductController = (async (req, res) => {
   try {
-    const { pid, cid } = req.params;
+    const { pid, cid } = req.params || {};
+    if (!pid || !cid) {
+      return res.status(400).send({
+        success: false,
+        message: "Both pid and cid are required",
+      });
+    }
```

Comment:
Nit: I feel like this is not necessary since they are path params (not query params/request body) so they will definitely  always be defined

Reply:
But we can keep it for branch testing (though in application code it will never happen unless something is wrong with the integration)